ui/app/js/controllers/permission_scheduled_change_new_controller.js

```diff
@@ -0,0 +1,241 @@
+/*global sweetAlert swal */
+angular.module('app').controller('NewPermissionScheduledChangeCtrl',
+function ($scope, $modalInstance, CSRF, Permissions, scheduled_changes, sc) {
+
+  $scope.loading = true;
+  $scope.scheduled_changes = scheduled_changes;
+  $scope.scheduledpermissions = [];
+  $scope.currentItemTab = 1;
+  $scope.is_edit = true;
+  $scope.sc = angular.copy(sc);
+  $scope.permission = {
+    permission: '',
+    options_as_json: ''
+  };
+  $scope.errors = {
+    permissions: {}
+  };
+
+  $scope.sc.permissions = [];
+  Permissions.getUserPermissions(sc.username)
+  .then(function(permissions) {
+    _.forEach(permissions, function(p) {
+      if (p.options) {
+        p.options_as_json = JSON.stringify(p['options']);
+      }
+    });
+    $scope.sc.permissions = permissions;
+    // console.log('$scope.sc.permissions');
+    // console.dbg($scope.sc.permissions);
+    $scope.loading = false;
+  });
+  $scope.sc.roles = [];
+  Permissions.getUserRoles(sc.username)
```

Comment:
I don't think you need to worry about getting Roles here - they're not used at all, right? Same thing below when you're getting the list of all of the Roles.